### ISSUE TITLE
Proposal to add CAR ID column as option for analysis grids

### ISSUE THREAD

**Issue by sksmi** [SACGF/variantgrid#187]
**Is your feature request related to a problem? Please describe.**
Would like to use clingen allele registry IDs to compare between hg19 & 38 analyses of the same vcf as part of testing, but don't think the CAR ID is available as a column as far as I can see.  Think it could also be useful to have in general, although haven't come across another specific use case.

**Describe the solution you'd like**
If it's quick & easy, would it possible to add CAR IDs as an additional column that can be added via custom columns in user settings?

**Describe alternatives you've considered**
If difficult, can probably devise an alternate approach for testing.

**Comment by davmlaw** [comment]
We only perform liftover (ie create alleles, contact ClinGen etc) when someone classifies a variant, or visits the variant detail page.

We have 250,858,853 variants in variantgrid.com and ClinGen only allow a max of 2000 variants per API call (ie 125k calls just from that server alone)

They do allow you to upload a VCF, so if we did some work to support that method perhaps we could populate a server's worth of variants - raise an issue about that if you'd like (but no chance it can make SA Path upgrade)

**Comment by davmlaw** [comment]
It's not hard to make a new column so I may do it. I'll just add a description saying it won't necessarily be populated.

But if you click on the variant link in an analysis, it'll open the variant details page, and then populate the CAR, so it won't take that long to be able to create a few.

If I combine it with #191 - to manually be able to liftover - should be able to at least make testing a bit easier..... it won't be easy so any help would be good

**Comment by davmlaw** [comment]
OK, added clingen allele column (not adding to any existing custom columns, need to manually add them yourself)

For admins, there are 2 new "populate clingen allele" buttons:

* On VCF page
* In analysis node "debug" tab 

Please only use this on VCF/nodes of say sub 10k variants - thanks!

**Comment by sksmi** [comment]
Thank you - so fantastic! Will save me weeks.  Did a quick test via analysis node debug tab on samples for hg37 & hg38. Works well & really interesting to see what does/doesn't map to CAR ids. 

Don't think we have any small vcfs to trial vcf page button. If it's easy, wondering if that one should be removed as think it would be pretty rare for our vcfs to have <10k variants? - worried about me or someone else accidentally clicking it. Or is it poss to add "warning: do not use if > 10k variants" or something like that as a reminder. We've got a fair few admin users now..

**Comment by davmlaw** [comment]
ok, will put a hard limit of 50k or something on VCFs

**Comment by davmlaw** [comment]
Added a new setting:

```CLINGEN_ALLELE_REGISTRY_MAX_MANUAL_REQUESTS = 10_000  ```

So it won't go nuts and use up a worker DDOSing ClinGen with 500 calls.

Made a little tooltip on the button

**Comment by sksmi** [comment]
Have tested a couple of times. Looks good, waiting on error 500 resolution to sign-off on testing.

**Comment by davmlaw** [comment]
Error was serializing a GeneListNode:

```
  File "/home/dlawrence/localwork/variantgrid/genes/serializers.py", line 49, in get_can_write
    user = self.context['request'].user
KeyError: 'request'
```

Fixed and patched VG test

**Comment by sksmi** [comment]
PASSED. https://app.qase.io/run/VG/dashboard/30

@davmlaw Minor comments (no action required):
- some c.HGVS & p.HGVS nomenclature not represented in standard format (will log as separate issue)
- some differences in choice of REP/Canonical transcript (this is formally tested elsewhere, no major impact)
- identified known AF issue where HET variant AF reported as 100% due to loss of ref after normalisation
- identified known ClinGen issue with HGVS normalisation
